HumanEvalLean/HumanEval106.lean

```diff
@@ -1,5 +1,207 @@
-def f : Unit :=
-  ()
+module
+
+import Std.Tactic.Do
+-- Sadly, it's apparently currently impossible to easily prove the size of a
+-- `Nat` range without unfolding internal stuff. This should be fixed in the standard library.
+import all Init.Data.Iterators.Consumers.Loop
+
+open Std.Do
+
+def f (n : Nat) : List Nat := Id.run do
```

Comment:
I would be very interested in a faster version of `f` that builds an array instead of a list and avoids the nested loop by referring to `ret[i - 2]` as appropriate.